<?xml version="1.0"?>
<launch>
 
<param name="robot_description_board" command="$(find xacro)/xacro '$(find franka_cal_sim)/xacro/aprilgrid.xacro'" />

<include file="$(find franka_cal_sim)/launch/franka.launch">
</include>

<node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>

<!-- Spawn the checkbaord into Gazebo -->


<node name="image_view" pkg="image_view" type="image_view" args="image:=/simple_camera/image_raw"/>

<node name="rl_client" pkg="franka_cal_sim" type="RL_agent.py" output="screen">
<rosparam command="load" file="$(find franka_cal_sim)/config/config.yaml" />
</node>

<node name="estimator" pkg="franka_cal_sim" type="estimation_service_server_imu.py" output="screen">
</node>


<node pkg="nodelet" type="nodelet" name="simulation_server" args="load franka_cal_sim/simulation_server standalone_nodelet" output="screen">
</node>

<node pkg="nodelet" type="nodelet" name="model_client" args="load franka_cal_sim/model_client standalone_nodelet" output="screen">
</node>


<node name="spawn_sdf" pkg="gazebo_ros" type="spawn_model" args="-file $(find franka_cal_sim)/models/checkerboard/checkerboard_8_7_0_06/checkerboard.sdf -sdf -model checkerboard -x -2 -y -0.35 -z 0.5 -P 1.57 -unpause" />

<node name="rosbag_srv" pkg="rosbag_recorder" type="rosbag_recorder_server.py" output="screen">
</node>


</launch>
